src/lib/doc.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ManuscriptNode,
+  ManuscriptTransaction,
+  schema,
+} from '@manuscripts/transform'
+import { findChildrenByType } from 'prosemirror-utils'
+
+export const insertSupplementsNode = (tr: ManuscriptTransaction) => {
+  const doc = tr.doc
+  const supplements = findChildrenByType(doc, schema.nodes.supplements)[0]
+  if (supplements) {
+    return supplements.pos
+  }
+  const abstracts = findAbstractsNode(doc)
```

Comment:
is this correct to put supplements into abstracts?

Reply:
The insert pos is abstracts pos - 1, that should be before abstracts, shouldn't it?

Reply:
Right, but I mean we don't display it at all, shouldn't it be at the end of the doc or in the meta section? Or are we mimickng JATS structure in that aspect?

Reply:
I agree that it should be at the end, but this is how the current schema is and I didn't want to change it here. Once the migrations work is done, we can update the schema and move the supplements to the end.

Reply:
got it, thanks for the info